ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt

```diff
@@ -60,35 +68,37 @@ class DataCollectionFragment : AbstractFragment(), BackPressListener {
     viewModel.loadSubmissionDetails(args)
     viewModel.submission.observe(viewLifecycleOwner) { submission: Loadable<Submission> ->
       submission.value().ifPresent {
-        viewPager.adapter = DataCollectionViewPagerAdapter(this, it.job.tasksSorted)
+        viewPager.adapter = viewPagerAdapterFactory.create(this, it.job.tasksSorted, viewModel)
       }
     }
 
+    viewModel.currentPosition.observe(viewLifecycleOwner) { viewPager.currentItem = it }
+
+    viewModel.continueResults
+      .observeOn(schedulers.ui())
+      .`as`(autoDisposable(viewLifecycleOwner))
+      .subscribe { error -> handleContinueClickError(error) }
+
     binding.viewModel = viewModel
-    binding.dataCollectionContinueButton.setOnClickListener { onNextClick() }
     binding.lifecycleOwner = this
 
-    (activity as MainActivity?)?.let {
-      it.setActionBar(binding.dataCollectionToolbar, showTitle = false)
-    }
+    (activity as AbstractActivity?)?.setActionBar(binding.dataCollectionToolbar, showTitle = false)
 
     return binding.root
   }
 
-  override fun onBack(): Boolean {
-    return if (viewPager.currentItem == 0) {
+  override fun onBack(): Boolean =
+    if (viewPager.currentItem == 0) {
       // If the user is currently looking at the first step, allow the system to handle the
       // Back button. This calls finish() on this activity and pops the back stack.
       false
     } else {
       // Otherwise, select the previous step.
-      viewPager.currentItem = viewPager.currentItem - 1
+      viewModel.currentPosition.value = viewModel.currentPosition.value!! - 1
```

Comment:
The ViewModel needs to keep track of the current position so that we can keep track of which task is active, currently this is used in onContinueClicked to get validate the results of the current task. Then the ViewPager item is updated by the DataCollectionFragments currentPosition observer.

Reply:
acked